### PR TITLE
feat: support multichannel SSE connections

### PR DESCRIPTION
**Pull request by klausborges**
## What does PR do?

Support multi-channel connections on SSE, which is in a newer version of the SSE service.

This PR **changes the function signature** for `openSSEListening` for a few reasons:

- Support a single channel or multiple channels as the first parameter (URL is built differently for multi-channel)
- Move `token`/`region` to second parameter since it's decoupled from the channels

Type definitions show these changes in the editor properly in case of updating from a previous version.

### Example

```ts
const single = await openSSEListening(
  { channel: "device_inspector", resource_id: "1234" },
  { token: "my-token", region: "usa-1" },
);

const multi = await openSSEListening(
  [
    { channel: "device_inspector", resource_id: "1234" },
    { channel: "device_inspector", resource_id: "5678" },
    { channel: "ui" },
  ],
  { token: "my-token", region: "usa-1" },
);
```

For comparison, here's an example using the **old function signature**:

```ts
const single = await openSSEListening({
  channel: "device_inspector",
  resource_id: "1234",
  token: "my-token",
  region: "usa-1",
});
```

## Type of alteration

- [ ] Bug fix
- [x] New feature
- [x] Breaking change
- [ ] Documentation update

### CODE REVIEW

File: src/infrastructure/apiSSE.ts [hunk 1]
```diff
@@ -1,19 +1,19 @@
 import { GenericModuleParams } from "../common/TagoIOModule";
 import regions from "../regions";
 
-const channelsWithID = ["device_inspector", "analysis_console", "ui_dashboard"] as const;
-const channelsWithoutID = ["notification", "analysis_trigger", "ui"] as const;
+const channelsWithID = ["analysis_console", "device_inspector", "device_data", "ui_dashboard"] as const;
+const channelsWithoutID = ["analysis_trigger", "notification", "ui"] as const;
```
Comment (klausborges) [inline src/infrastructure/apiSSE.ts:5]: Added `device_data`, ordered alphabetically, also checked if these are all supported channels

File: src/infrastructure/apiSSE.ts [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
 const channels = [...channelsWithID, ...channelsWithoutID] as const;
 
 type ChannelWithID = (typeof channelsWithID)[number];
 type ChannelWithoutID = (typeof channelsWithoutID)[number];
 
-type OpenSSEWithID = GenericModuleParams & {
```
Comment (klausborges) [inline src/infrastructure/apiSSE.ts:11]: Moved `GenericModuleParams` union to the function parameter type

File: src/infrastructure/apiSSE.ts [hunk 3]
```diff
@@ -32,25 +32,42 @@ async function loadEventSourceLib(): Promise<typeof EventSource> {
   }
 }
 
-async function openSSEListening(params: OpenSSEConfig): Promise<EventSource> {
+async function createEventSource(url: URL) {
```
Comment (klausborges) [inline src/infrastructure/apiSSE.ts:35]: Split these 2 functions for reuse, same logic as before for this part

File: src/infrastructure/apiSSE.ts [hunk 4]
```diff
@@ -32,25 +32,42 @@ async function loadEventSourceLib(): Promise<typeof EventSource> {
...
+  const { channel } = configuration;
+
+  if (isChannelWithID(configuration)) {
+    return `${channel}.${configuration.resource_id}`;
+  }
+
+  return channel;
+}
+
+async function openSSEListening(params: OpenSSEConfig & GenericModuleParams): Promise<EventSource> {
```
Comment (klausborges) [inline src/infrastructure/apiSSE.ts:50]: Still works exactly the same as before

File: src/infrastructure/apiSSE.ts (hunk 2 above)
Comment (klausborges) [inline src/infrastructure/apiSSE.ts:11]: `token`/`region` (from `GenericModuleParams`) have been moved to the second parameter of `openSSEListening`

**Review by felipefdl — APPROVED** [review]

**Pull request merged**